src/NHibernate/Proxy/DefaultDynamicProxyMethodCheckerExtensions.cs

```diff
@@ -47,5 +45,11 @@ private static bool IsDisposeMethod(MethodInfo method)
 			return method.Name.Equals("Dispose") && method.MemberType == MemberTypes.Method && method.GetParameters().Length == 0;
 			// return method.Name.Equals("Dispose") && method.IsMethodOf(typeof(IDisposable));
 		}
+
+		private static bool IsFinalizeMethod(MethodInfo method)
+		{
+			return "finalize".Equals(method.Name, StringComparison.OrdinalIgnoreCase) &&
```

Comment:
Why this method name check? Seems not needed.

Reply:
It was in the original code, so I left it here.

Reply:
But it's not original check anymore so let's remove it as obsolete.